src/server/scripts/Events/hallows_end.cpp

```diff
@@ -57,13 +78,391 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                return;
+
+            handled = true;
+            if (Player* target = GetHitPlayer())
+                GetCaster()->CastSpell(target, SPELL_CREATE_BUCKET, true);
+            else if (Unit* tgt = GetHitUnit())
+                GetCaster()->CastSpell(tgt, SPELL_WATER_SPLASH, true);
+        }
+
+        void Register() override
+        {
+            OnEffectHitTarget += SpellEffectFn(spell_hallows_end_bucket_lands_SpellScript::HandleDummy, EFFECT_0, SPELL_EFFECT_DUMMY);
+        }
+    };
+
+    SpellScript* GetSpellScript() const override
+    {
+        return new spell_hallows_end_bucket_lands_SpellScript();
+    }
+};
+
+class spell_hallows_end_base_fire : public SpellScriptLoader
+{
+public:
+    spell_hallows_end_base_fire() : SpellScriptLoader("spell_hallows_end_base_fire") { }
+
+    class spell_hallows_end_base_fire_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_hallows_end_base_fire_AuraScript);
+
+        void HandleEffectPeriodicUpdate(AuraEffect* /*aurEff*/)
+        {
+            if (Unit* owner = GetUnitOwner())
+            {
+                if (Aura* aura = owner->GetAura(SPELL_SMALL_FIRE))
+                    if (aura->GetStackAmount() < 20)
+                        owner->CastSpell(owner, SPELL_SMALL_FIRE, true);
+
+                if (Aura* aura = owner->GetAura(SPELL_BIG_FIRE))
+                    if (aura->GetStackAmount() < 20)
+                        owner->CastSpell(owner, SPELL_BIG_FIRE, true);
+            }
+       }
+
+        void HandleEffectApply(AuraEffect const* aurEff, AuraEffectHandleModes /*mode*/)
+        {
+            if (Unit* owner = GetUnitOwner())
+            {
+                if (roll_chance_i(50))
+                    owner->SetAuraStack(SPELL_BIG_FIRE, owner, 20);
+                else
+                    owner->SetAuraStack(SPELL_SMALL_FIRE, owner, 20);
+            }
+        }
+
+        void Register() override
+        {
+            OnEffectUpdatePeriodic += AuraEffectUpdatePeriodicFn(spell_hallows_end_base_fire_AuraScript::HandleEffectPeriodicUpdate, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+            OnEffectApply += AuraEffectApplyFn(spell_hallows_end_base_fire_AuraScript::HandleEffectApply, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY, AURA_EFFECT_HANDLE_REAL);
+        }
+    };
+
+    AuraScript* GetAuraScript() const override
+    {
+        return new spell_hallows_end_base_fire_AuraScript();
+    }
+};
+
+class npc_costumed_orphan_matron : public CreatureScript
+{
+public:
+    npc_costumed_orphan_matron() : CreatureScript("npc_costumed_orphan_matron") { }
+
+    struct npc_costumed_orphan_matronAI : public ScriptedAI
+    {
+        npc_costumed_orphan_matronAI(Creature* creature) : ScriptedAI(creature) {}
+
+        void Reset()
+        {
+            _events.ScheduleEvent(EVENT_BEGIN, /*Minutes(30)*/Seconds(5));
+        }
+
+        void GetInitXYZ(float &x, float &y, float &z, float &o, uint32 &path)
```

Comment:
should make this helper return boolean.
u know, return false in case the path passed is not handled in the switch